### ISSUE TITLE
add support for custom error messages on failure

### ISSUE THREAD

**Issue by jasoncorbett** [ActivisionGameScience/assertpy#35]
In order to have a better context in the stack trace, it would be helpful if you could provide a name property to the value to handle assertions better.  For example:

```py
assert_that(1 + 2).named('the result of 1 + 2').is_equal_to(4)
```

will raise error message:

```
Expected [the result of 1 + 2] <3> to be equal to <4>, but was not.
```

or maybe:

```
Expected [the result of 1 + 2] to be equal to <4>, but it was <3>.
```

**Comment by saturnboy** [comment]
AssertJ uses `describedAs()`, and more commonly its alias `as()`, to add an extra text description to the exception message.

In `assertpy`, we should probably mirror this behavior, but use `as()`, with an alias to `described_as()`, to add extra text to an error message.  For example:

```py
assert_that(1).is_equal_to(2)
```

Outputs:

```
AssertionError: Expected <1> to be equal to <2>, but was not.
```

Including a custom error description:

```py
assert_that(1).as('i like cheese').is_equal_to(2)
```

Should output something like this:

```
AssertionError: [i like cheese] Expected <1> to be equal to <2>, but was not.
```

**Comment by jasoncorbett** [comment]
I am fine with the prefixing method, my main requirement is the customization for clarity.  What you proposed seems to be able to do that.

**Comment by saturnboy** [comment]
Thanks for this suggestion, `described_as()` is now available in v0.8

pip here:
https://pypi.python.org/pypi/assertpy/0.8

docs here:
https://github.com/ActivisionGameScience/assertpy#custom-error-messages
